runtime/nvqir/NVQIR.cpp

```diff
@@ -184,15 +184,49 @@ Array *__quantum__rt__qubit_allocate_array(uint64_t size) {
   return vectorSizetToArray(qubitIdxs);
 }
 
-Array *__quantum__rt__qubit_allocate_array_with_state_fp64(
+Array *__quantum__rt__qubit_allocate_array_with_state_complex32(
+    uint64_t size, std::complex<float> *data);
+
+Array *__quantum__rt__qubit_allocate_array_with_state_complex64(
     uint64_t size, std::complex<double> *data) {
-  ScopedTraceWithContext("NVQIR::qubit_allocate_array_with_data_fp64", size);
+  ScopedTraceWithContext("NVQIR::qubit_allocate_array_with_data_complex64",
+                         size);
   __quantum__rt__initialize(0, nullptr);
+  if (nvqir::getCircuitSimulatorInternal()->isDoublePrecision()) {
+    auto qubitIdxs = nvqir::getCircuitSimulatorInternal()->allocateQubits(
+        size, data, cudaq::simulation_precision::fp64);
+    return vectorSizetToArray(qubitIdxs);
+  }
+  // Input was complex<double> but we prefer complex<float>. Make a copy,
+  // truncating the values.
+  auto convertData = std::make_unique<std::complex<float>[]>(size);
```

Comment:
I think it would be good to add some `cudaq::info(...)` messages when you make these copies

Reply:
Sure.